flink-kubernetes/src/test/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClientTest.java

```diff
@@ -411,6 +432,16 @@ void testStopAndCleanupCluster() throws Exception {
                 .isEmpty();
     }
 
+    @Test
+    void testWatchPodsAndDoCallback() throws Exception {
+        mockPodEventWithLabels(TESTING_LABELS);
+        // the count latch for events.
+        final CountDownLatch eventLatch = new CountDownLatch(4);
+        this.flinkKubeClient.watchPodsAndDoCallback(
```

Comment:
It's better to check each event is received.

Maybe you can use three CountDownLatch and a anonymous classes here

Reply:
It seems that we have too many class implements `FlinkKubeClient.WatchCallbackHandler<T>` only for testing purpose. We'd better introduce a more general and reusable `TestingKubernetesPodCallbackHandler` as the first commit and rewrite all other impls like `NoOpWatchCallbackHandler` & `TestingCallbackHandler`. 

Don't worry too much, if you don't know how to do this refactor, I can push this commit as example.

Reply:
I agreed that we should also check all received events. IMH, We can introduce three `CompletableFuture<Watch.Action>` to handle this.